main/saturn-node.js

```diff
@@ -152,12 +152,15 @@ async function start (/** @type {Context} */ ctx) {
     console.log(msg)
     appendToChildLog(msg)
     ctx.recordActivity({ source: 'Saturn', type: 'info', message: msg })
-    Sentry.captureException(new Error(msg), scope => {
+
+    ready = false
+  })
+
+  childProcess.on('close', () => {
+    Sentry.captureException('Saturn node exited', scope => {
```

Comment:
Hmm, this is not going to tell us whether the Saturn node exited successfully (exit code 0 - I think we should not report this BTW), failed with an error (non-zero exit code) or crashed on something like SIGSEGV.

I think we should capture the exit reason in `childProcess.on('exit')` handler and then report that exit reason to Sentry.

```ts
// on('exit')
moduleExitReason = signal || code ? reason : null

// on('close')
if (moduleExitReason) {
  const msg = `Saturn node exited ${moduleExitReason}`
  Sentry.captureException(msg, scope => {
   // ...
  })
}
```

WDYT?

Reply:
Also inside `start()` function:

```ts
moduleExitReason = null // or undefined?
childLog = []
appendToChildLog('Starting Saturn node')
// etc.
```

Reply:
> Hmm, this is not going to tell us whether the Saturn node exited successfully (exit code 0 - I think we should not report this BTW), failed with an error (non-zero exit code) or crashed on something like SIGSEGV.

Is a 0 exit code actually accepted? I believe this should never happen.

Otherwise I'm +1 to adding this change

Reply:
> Is a 0 exit code actually accepted? I believe this should never happen.

It happens when the node shuts down cleanly. In theory, this should be the case when the Station exits. In practice, I am not sure if the current implementation in Station is waiting for all modules to stop before exiting the app.

When emulate this scenario by running `ps ax | grep saturn` and then `kill <pid-of-saturn-L2-node>`, I get the following Activity Log entry:

<img width="791" alt="Screenshot 2022-11-23 at 15 30 27" src="https://user-images.githubusercontent.com/1140553/203575652-383af9cb-7f5f-4255-a6ba-b960d7aa1a7b.png">

Reply:
> When emulate this scenario by running `ps ax | grep saturn` and then `kill <pid-of-saturn-L2-node>`, I get the following Activity Log entry:

If saturn was `kill`ed, we'd definitely want to know, right? Otoh, it would be really nice if we could just rely on `0` being an ok exit code and us not having to do anything. Otherwise, if `0` can also mean trouble, we would always need to treat it as an error except if Station is exiting (remove `"close"` listener on before exit?). WDYT?

Reply:
> If saturn was `kill`ed, we'd definitely want to know, right? 

Hmm, you are right. We do want to know when saturn was `kill`ed.

> Otoh, it would be really nice if we could just rely on `0` being an ok exit code and us not having to do anything. Otherwise, if `0` can also mean trouble, we would always need to treat it as an error except if Station is exiting (remove `"close"` listener on before exit?). WDYT?

I think for now, let's always treat exit code `0` as a trouble and report it to Sentry.

For longer term, let's improve Station's shutdown sequence to wait for all modules to exit, ignore exit code 0 and report non-zero exit codes as errors. That would be a follow-up issue to be created.

For this PR, the remaining work is to save `moduleExitReason` in `exit` listener and include it in the error message reported to Sentry from `close` listener. So that we know whether the Saturn module exited cleanly or crashed.

WDYT?

Reply:
sounds good all! like this?